App/screens/CredentialOffer.tsx

```diff
@@ -31,123 +34,187 @@ const styles = StyleSheet.create({
   container: {
     flex: 1,
     flexDirection: 'column',
-    backgroundColor: CredentialOfferTheme.background,
+    backgroundColor: ColorPallet.brand.primaryBackground,
+  },
+  listItem: {
+    paddingHorizontal: 25,
+    paddingTop: 16,
+    backgroundColor: ColorPallet.brand.secondaryBackground,
+  },
+  label: {
+    ...TextTheme.label,
+  },
+  textContainer: {
+    justifyContent: 'space-between',
+    minHeight: TextTheme.normal.fontSize,
+    paddingVertical: 15,
+    height: 90,
+  },
+  text: {
+    ...TextTheme.normal,
   },
 })
 
 const CredentialOffer: React.FC<CredentialOfferProps> = ({ navigation, route }) => {
+  if (!route || !route.params) {
+    throw new Error('CredentialOffer route prams were not set properly')
+  }
+
+  const { credentialId } = route.params
   const { agent } = useAgent()
   const { t } = useTranslation()
+  const [, dispatch] = useContext(Context)
   const [buttonsVisible, setButtonsVisible] = useState(true)
   const [pendingModalVisible, setPendingModalVisible] = useState(false)
   const [successModalVisible, setSuccessModalVisible] = useState(false)
   const [declinedModalVisible, setDeclinedModalVisible] = useState(false)
+  const dateFormatOptions = { year: 'numeric', month: 'short', day: 'numeric' }
+  const credential = useCredentialById(credentialId)
 
-  if (!agent?.credentials) {
-    Toast.show({
-      type: ToastType.Error,
-      text1: t('Global.Failure'),
-      text2: t('Global.SomethingWentWrong'),
-    })
-
-    navigation.goBack()
-    return null
+  if (!credential) {
+    throw new Error('Unable to fetch credential from AFJ')
   }
 
-  const getCredentialRecord = (credentialId?: string): CredentialRecord | void => {
-    try {
-      if (!credentialId) {
-        throw new Error(t('CredentialOffer.CredentialNotFound'))
-      }
-      return useCredentialById(credentialId)
-    } catch (e: unknown) {
-      Toast.show({
-        type: ToastType.Error,
-        text1: t('Global.Failure'),
-        text2: t('Global.SomethingWentWrong'),
-      })
-
-      navigation.goBack()
-    }
+  if (!agent) {
+    throw new Error('Unable to fetch agent from AFJ')
   }
 
-  const credential = getCredentialRecord(route?.params?.credentialId)
+  const { name: schemaName } = parsedSchema(credential)
+  // @ts-ignore next-line
+  const { invitation } = useConnectionById(credential.connectionId)
 
-  if (!credential) {
-    Toast.show({
-      type: ToastType.Error,
-      text1: t('Global.Failure'),
-      text2: t('CredentialOffer.CredentialNotFound'),
-    })
-    navigation.goBack()
-    return null
+  if (!invitation) {
+    throw new Error('Unable to invitation from AFJ')
```

Comment:
I would create a class called `AriesFrameworkError` or something like that to distinguish from `BifoldError` and some generic `Error`

Reply:
I think they should become `BifoldError` with title, message, and code. My though is if any of these errors are thrown the offer screen won't be usable to the end user so it should throw an error the user can see and then exit/pop() back to the Home Screen. Or maybe these things should be checked before the offer screen is even shown. Thoughts?

Reply:
The errors still are for developers but It would offer more granularity when logging ("was it a problem with the framework or the wallet itself?")

Reply:
I think unique "code" would help with that in the long run. In any case, I can create the new Error class.